### PR TITLE
Develop/fixes/td 788 welcome email links are invalid by the time

### PR DESCRIPTION
**Pull request by manishagarwalnhs**
### JIRA link
https://hee-tis.atlassian.net/browse/TD-788

### Description
Users report that welcome email links are invalid by the time they receive them for delegate users and normal registered user. This has been fixed now.

### Screenshots
https://hee-tis.atlassian.net/browse/TD-788

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors (see [info on Text Editor settings](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546185813/DLS+Dev+Process) to avoid whitespace changes)
- [x] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript
- [x] Tested any Views or partials created or changed with [Wave Chrome plugin](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh/related) and addressed any valid accessibility issues
- [ ] Updated/added documentation in [Confluence](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546939432/DLS+Code) and/or [GitHub Readme](https://github.com/TechnologyEnhancedLearning/DLSV2/blob/master/README.md). List of documentation links added/changed:
  - [doc_1_here](link_1_here)
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken
- [x] Scanned over my pull request in GitHub and addressed any warnings from the GitHub Build and Test checks.

### CODE REVIEW

File: DigitalLearningSolutions.Web.Tests/Controllers/VerifyEmailControllerTests.cs [hunk 1]
```diff
@@ -126,6 +129,7 @@ public void Index_with_unverified_email_for_unapproved_delegate_passes_centre_id
             const int userId = 1;
             const int centreIdForUnapprovedDelegate = 2;
             var now = new DateTime(2022, 1, 1);
+            //var now = DateTime.Now;
```
Comment (martinfoster-hee) [inline DigitalLearningSolutions.Web.Tests/Controllers/VerifyEmailControllerTests.cs:133]: Could probably delete the commented code here.

File: DigitalLearningSolutions.Web/Controllers/Register/ClaimAccountController.cs [hunk 2]
```diff
@@ -270,6 +280,15 @@ public IActionResult AccountAlreadyExists(string email, string centreName)
...
 
+            var emailOutData = emailService.GetEmailOutUsingEmail(model.Email);
+
+            if (emailOutData != null && (emailOutData.DeliverAfter + EmailVerificationHashLifetime < clockUtility.UtcNow))
+            {
+                return View("VerificationLinkExpired");
+                
+            }
+
+
```
Comment (martinfoster-hee) [inline DigitalLearningSolutions.Web/Controllers/Register/ClaimAccountController.cs:291]: Maybe tidy up the whitespace here? :)

**Review by martinfoster-hee — COMMENTED** [review]
A couple of (tiny) formatting tweaks but ok otherwise.

File: DigitalLearningSolutions.Data.Migrations/Properties/Resources.resx [hunk 3]
```diff
@@ -223,4 +223,10 @@
   <data name="UAR_859_PopulateUsersTableFromAccountsTables_DOWN" type="System.Resources.ResXFileRef, System.Windows.Forms">
     <value>..\Scripts\UAR-859-PopulateUsersTableFromAccountsTables-DOWN.sql;System.String, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089;Windows-1252</value>
   </data>
-</root>
\ No newline at end of file
+  <data name="TD-788_AddResetExpiryDateTime" type="System.Resources.ResXFileRef, System.Windows.Forms">
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Data.Migrations/Properties/Resources.resx:226]: Not sure these are needed? See other comments.

File: DigitalLearningSolutions.Data.Migrations/RecurringSqlJobs/DeleteEmailVerificationHashesRecordsOlderThan4Days.sql [hunk 4]
```diff
@@ -0,0 +1,5 @@
+﻿-- Deletes EmailVerificationHashes records older than 4 days. Should be run every night at 2:02AM.
+BEGIN
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Data.Migrations/RecurringSqlJobs/DeleteEmailVerificationHashesRecordsOlderThan4Days.sql:2]: We are deleting EmailVerificationHashes that are more than 4 days old every day and checking elsewhere in the code that the hash created date isn't more than 14 days old?

**Review by kevwhitt-hee — COMMENTED** [review]
@manishagarwalnhs, I think we need to discuss this one. We appear to have introduced a ResetExpiryDateTime to the ResetPassword table but then checking the EmailVerificationHashes.CreatedDate to determine whether a link is valid.

**Comment by manishagarwalnhs** [comment]
Not Required as per Kevin's feedback

**Pull request closed without merging**